tools/partutil/extend_partition.go

```diff
@@ -0,0 +1,136 @@
+package partutil
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+)
+
+//ExtendPartition extends a partition to a specific end sector
+func ExtendPartition(disk string, partNumInt, end int) error {
+	if len(disk) <= 0 || partNumInt <= 0 {
+		return errors.New("invalid disk name or partition number")
+	}
+
+	//for cases like loop5p1
+	partNum := strconv.Itoa(partNumInt)
+	if disk[len(disk)-1] >= '0' && disk[len(disk)-1] <= '9' {
+		partNum = "p" + partNum
+	}
+
+	partName := disk + partNum
+
+	//dump partition table to a file
+	cmd := string("sudo sfdisk --dump ") + disk + " > extend_partition_tmp"
```

Comment:
As this is partition table, I would recommend to name is: partition_table
We can get rid of *_tmp I think